Impossible to build for windows

Hi,

In an application I use cloudflare/tableflip lib.

I've got an issue when I try to build for windows/386:

```

# github.com/cloudflare/tableflip
../../../../go/pkg/mod/github.com/cloudflare/tableflip@v1.0.0/env.go:13:2: cannot use syscall.CloseOnExec (type func(syscall.Handle)) as type func(int) in field value
../../../../go/pkg/mod/github.com/cloudflare/tableflip@v1.0.0/fds.go:344:36: not enough arguments in call to syscall.Syscall
../../../../go/pkg/mod/github.com/cloudflare/tableflip@v1.0.0/fds.go:344:37: undefined: syscall.SYS_FCNTL
../../../../go/pkg/mod/github.com/cloudflare/tableflip@v1.0.0/fds.go:344:60: undefined: syscall.F_DUPFD_CLOEXEC
```

Thanks

--- a/README.md
+++ b/README.md
@@ -16,47 +16,37 @@ has the following goals:
 * Crashing during initialisation is OK
 * Only a single upgrade is ever run in parallel
 
-`tableflip` does not work on Windows.
+**`tableflip` works on Linux and macOS.**
 
-It's easy to get started:
+## Using the library
 
 ```Go
-upg, err := tableflip.New(tableflip.Options{})
-if err != nil {
-	panic(err)
-}
+upg, _ := tableflip.New(tableflip.Options{})
 defer upg.Stop()
 
 go func() {
 	sig := make(chan os.Signal, 1)
 	signal.Notify(sig, syscall.SIGHUP)
 	for range sig {
-		err := upg.Upgrade()
-		if err != nil {
-			log.Println("Upgrade failed:", err)
-			continue
-		}
-
-		log.Println("Upgrade succeeded")
+		upg.Upgrade()
 	}
 }()
 
-ln, err := upg.Fds.Listen("tcp", "localhost:8080")
-if err != nil {
-	log.Fatalln("Can't listen:", err)
-}
+// Listen must be called before Ready
+ln, _ := upg.Listen("tcp", "localhost:8080")
+defer ln.Close()
 
-var server http.Server
-go server.Serve(ln)
+go http.Serve(ln, nil)
 
 if err := upg.Ready(); err != nil {
 	panic(err)
 }
+
 <-upg.Exit()
-
-time.AfterFunc(30*time.Second, func() {
-	os.Exit(1)
-})
-
-_ = server.Shutdown(context.Background())
 ```
+
+Please see the more elaborate [graceful shutdown with net/http](http_example_test.go) example.
+
+## Logging to `journald`
+
+The logs of a process using `tableflip` may go missing due to a [bug in journald](https://github.com/systemd/systemd/issues/13708). You can work around this by logging directly to journald, for example by using [go-systemd/journal](https://godoc.org/github.com/coreos/go-systemd/journal) and looking for the [$JOURNAL_STREAM](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#$JOURNAL_STREAM) environment variable.

--- a/http_example_test.go
+++ b/http_example_test.go
@@ -45,7 +45,8 @@ func Example_httpShutdown() {
 		}
 	}()
 
-	ln, err := upg.Fds.Listen("tcp", *listenAddr)
+	// Listen must be called before Ready
+	ln, err := upg.Listen("tcp", *listenAddr)
 	if err != nil {
 		log.Fatalln("Can't listen:", err)
 	}

